VisualProfiler.cs

```diff
@@ -1,747 +1,1002 @@
+                {
+                    temp = buffer[startIndex];
+                    buffer[startIndex] = buffer[endIndex];
+                    buffer[endIndex] = temp;
+                }
+            }
+
+            return bufferIndex;
+        }
+
+        private static int FtoA(float value, int displayedDecimalDigits, char[] buffer, int bufferIndex)
+        {
+            int integerDigits = (int)value;
+            int fractionalDigits = (int)((value - integerDigits) * Mathf.Pow(10.0f, displayedDecimalDigits));
+
+            bufferIndex = ItoA(integerDigits, buffer, bufferIndex);
+
+            if (displayedDecimalDigits != 0)
+            {
+                buffer[bufferIndex++] = '.';
+            }
+
+            if (fractionalDigits != 0)
+            {
+                bufferIndex = ItoA(fractionalDigits, buffer, bufferIndex);
+            }
+            else
+            {
+                for (int i = 0; i < displayedDecimalDigits; ++i)
+                {
+                    buffer[bufferIndex++] = '0';
+                }
+            }
+
+            return bufferIndex;
+        }
+
+        private static float AppFrameRate
+        {
+            get
+            {
+                // If the current XR SDK does not report refresh rate information, assume 60Hz.
+                float refreshRate = UnityEngine.XR.XRDevice.refreshRate;
+                return ((int)refreshRate == 0) ? 60.0f : refreshRate;
+            }
+        }
+
+        private static void AverageFrameTiming(FrameTiming[] frameTimings, uint frameTimingsCount, out float cpuFrameTime, out float gpuFrameTime)
+        {
+            double cpuTime = 0.0f;
+            double gpuTime = 0.0f;
+
+            for (int i = 0; i < frameTimingsCount; ++i)
+            {
+                cpuTime += frameTimings[i].cpuFrameTime;
+                gpuTime += frameTimings[i].gpuFrameTime;
+            }
+
+            cpuTime /= frameTimingsCount;
+            gpuTime /= frameTimingsCount;
+
+            cpuFrameTime = (float)(cpuTime * 0.001);
+            gpuFrameTime = (float)(gpuTime * 0.001);
+        }
+
+        private static ulong AppMemoryUsage
+        {
+            get
+            {
+#if WINDOWS_UWP
+                return MemoryManager.AppMemoryUsage;
+#else
+                return (ulong)Profiler.GetTotalAllocatedMemoryLong();
+#endif
+            }
+        }
+
+        private static ulong AppMemoryUsageLimit
+        {
+            get
+            {
+#if WINDOWS_UWP
+                return MemoryManager.AppMemoryUsageLimit;
+#else
+                return ConvertMegabytesToBytes(SystemInfo.systemMemorySize);
+#endif
+            }
+        }
+
+        private static bool WillDisplayedVertexCountDiffer(long oldCount, long newCount, int displayedDecimalDigits)
+        {
+            float oldCountK = oldCount / 1000.0f;
+            float newCountK = newCount / 1000.0f;
+            float decimalPower = Mathf.Pow(10.0f, displayedDecimalDigits);

```

Comment:
This is a SUPER perf nit, but I'm pretty sure you could divide `decimalPower` by `1000.0f` instead of both `oldCount` and `newCount` to save on a division operation, since you're just multiplying them together below anyway 😬

Reply:
Nice catch! Updated 🌟